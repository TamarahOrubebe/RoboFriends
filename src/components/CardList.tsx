import React from "react";
import Card from "./Card";

import { IRobot } from "./MainPage";

interface IProp {
  robots: Array<IRobot>;
}

const CardList = ({ robots }: IProp) => {
  return (
    <div>
      {robots.map((robot, i) => (
        <Card key={i} name={robot.name} email={robot.email} id={robot.id} />
      ))}
    </div>
  );
};

export default CardList;
