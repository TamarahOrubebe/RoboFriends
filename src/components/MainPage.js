import React, { Component } from "react";
import "../containers/App.css";
import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundary from "./ErrorBoundary";
import CounterButton from "./CounterButton";



class MainPage extends Component {


    componentDidMount() {
        this.props.handleFetch();
    }

  

    filterRobots = () => {
        return this.props.robots.filter((robot) => {
            return robot.name.toLowerCase().includes(this.props.searchfield.toLowerCase());
        });
    }
    

    render() {

      const {handleSearch, isPending } = this.props;

       
				

        return (
            <div className="tc">
                <h1 className="light-green f1">ROBOTIC FRIENDS</h1>
                <CounterButton />
                <SearchBox onSearch={handleSearch} />
                <Scroll id="scroll">

                    {isPending? <h1>Loading</h1> : <ErrorBoundary>
                        <CardList robots={this.filterRobots()} />
                    </ErrorBoundary> }
                    
                </Scroll>
            </div>
        );
    }
}
export default MainPage;
