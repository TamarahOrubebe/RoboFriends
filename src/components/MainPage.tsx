import React, { Component } from "react";
import "../containers/App.css";
import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundary from "./ErrorBoundary";
import CounterButton from "./CounterButton";

export interface IRobot{
    name: string,
    id: number,
    email: string
}

interface IProps{
    robots: Array<IRobot>,
    searchfield: string,
    handleFetch: VoidFunction,
    handleSearch: VoidFunction,
    isPending: boolean
}


interface IState{

}


class MainPage extends Component<IProps, IState> {

   
    componentDidMount(): void {
        this.props.handleFetch();
    }

  

    filterRobots = () => {
        return this.props.robots.filter((robot) => {
            return robot.name.toLowerCase().includes(this.props.searchfield.toLowerCase());
        });
    }
    

    render() {

      const {handleSearch, isPending } = this.props;

       
				

        return (
            <div className="tc">
                <h1 className="light-green f1">ROBOTIC FRIENDS</h1>
                <CounterButton color="red" />
                <SearchBox onSearch={handleSearch} />
                <Scroll>

                    {isPending? <h1>Loading</h1> : <ErrorBoundary>
                        <CardList robots={this.filterRobots()} />
                    </ErrorBoundary> }
                    
                </Scroll>
            </div>
        );
    }
}
export default MainPage;
