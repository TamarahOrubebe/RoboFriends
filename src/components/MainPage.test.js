import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";
import CardList from './CardList';

	
let wrapper;



it('renders MainPage without crashing', () => {

	const mockProps = {
		handleFetch: jest.fn(),
		robots: [],
		searchfield: "",
		isPending: false,
	}

	wrapper = shallow(<MainPage {...mockProps} />);
	expect(wrapper).toMatchSnapshot();
})



it('it filters robots correctly', () => {

	const mockProps2 = {
		handleFetch: jest.fn(),
		robots: [
			{
				id: 1,
				name: "John",
				email: "John@gmail.com",
			},
		],
		searchfield: "john",
		isPending: false,
	};

	const wrapper2 = shallow(<MainPage {...mockProps2} />);

	expect(wrapper2.instance().filterRobots()).toEqual([
		{
			id: 1,
			name: "John",
			email: "John@gmail.com"
		}
	]);

	
})




