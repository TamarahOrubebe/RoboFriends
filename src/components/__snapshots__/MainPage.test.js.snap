// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crashing 1`] = `
<div
  className="tc"
>
  <h1
    className="light-green f1"
  >
    ROBOTIC FRIENDS
  </h1>
  <CounterButton />
  <SearchBox />
  <Scroll
    id="scroll"
  >
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
